00_Base/src/interfaces/api/AbstractModuleApi.ts

```diff
@@ -199,13 +231,29 @@ export abstract class AbstractModuleApi<T extends IModule>
         reply.status(500).send(err);
       });
 
-    const _opts = {
+    const _opts: any = {
       method: httpMethod,
       url: this._toDataPath(namespace),
       schema: schema,
       handler: _handler,
     };
 
+    if (
+      !!schema &&
+      !!schema.headers &&
+      !!schema.headers.properties &&
+      !!schema.headers.properties.Authorization
+    ) {
+      _opts['preHandler'] = (this._server as unknown as any).auth([
+        (this._server as unknown as any).authorization,
+      ]);
+      if (!_opts['security']) {
```

Comment:
currently, nothing is using security, and when reviewing these few lines, we may not want to add the same AuthorazationSecurity in all cases where Authorization header is present, which can result in a bug. This code was created to help clunk the OCPI endpoints in one set and apply security across all of them. I am going to move this to as.ocpi.endpoint.ts class instead.